### ISSUE TITLE
Inbox counts one item, but shows empty

### ISSUE THREAD

**Issue by Riksu9000** [alainm23/planify#1254]
**Describe the bug**
I noticed I had one task in the inbox, so I opened it, but there were no tasks. If I added another task, it showed two tasks when there is only one.

**To Reproduce**
I have not yet figured out how to reproduce this issue

**Expected behavior**
The number displayed should match the number of items in the inbox.

**Desktop (please complete the following information):**
 - Fedora, planify 4.6

**Additional context**
I examined the database and here are the entries I think are relevant.

These are the inbox and the project the tasks were in
```
bfb56f6f-683c-4b31-b5d3-96f92b76b498|Inbox|blue|local|1|0|0|0|0|0|0|list|0||0|progress||0|||0|
f4a49b73-6c9e-4831-8b5b-7626a8da3bee|Kotona|blue|local|0|0|1|0|0|0|0|list|0||0|emoji|🏠️|0|||0|
```

Here are the problematic tasks. The second task is a subtask of the first task. As I understand it, one of the hashes indicates the project it is located in, and for some reason the subtask has the hash of the inbox.
```
fd9c29e4-0a2a-4d03-89cd-c127676bf0d4|***||{"date":"2024-04-23","timezone":"","isrecurring":false,"recurrencytype":"6","recurrencyinterval":"0","recurrencyweeks":"","recurrencycount":"0","recurrencyend":""}|2024-04-09T21:42:09+0300||2024-04-16T10:13:36+0300||f4a49b73-6c9e-4831-8b5b-7626a8da3bee||1|5|0|0|0|1|0||
4e83fcc7-0616-413b-b575-d81e2be7e11c|***||{"date":"2024-04-19","timezone":"","isrecurring":false,"recurrencytype":"6","recurrencyinterval":"0","recurrencyweeks":"","recurrencycount":"0","recurrencyend":""}|2024-04-16T10:13:04+0300||||bfb56f6f-683c-4b31-b5d3-96f92b76b498|fd9c29e4-0a2a-4d03-89cd-c127676bf0d4|1|0|0|0|0|0|0|79a30174-6dfa-4b49-90b4-291e09f796aa|
```

After finding this, I moved the subtask around, but only after restarting the app did the Inbox get fixed.

**Comment by Riksu9000** [comment]
I wonder if the project a subtask is located in needs to be stored at all, since it could be derived from the parent task. That way there are fewer ways to represent invalid states and it would avoid data redundancy errors like this, but would that make processing too slow?
